File: westcoast/src/main/java/frc/commands/ballhandling/Shooting.java
package frc.commands.ballhandling;

import edu.wpi.first.wpilibj.drive.DifferentialDrive;
import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
import edu.wpi.first.wpilibj2.command.WaitCommand;
import frc.robot.maps.RobotMap;
import frc.subsystems.BallHandlingSubsystem;
import frc.subsystems.vision.VisionRunner;
import frc.subsystems.vision.targeting.AlignToTarget;
import frc.subsystems.vision.targeting.StayOnTarget;

public class Shooting extends SequentialCommandGroup {

    public Shooting(RobotMap map, DifferentialDrive drive, VisionRunner vision, BallHandlingSubsystem ballHandler) {
        addCommands(
                new AlignToTarget(map, drive, vision),
                race(
                        new ShooterAccelerator(ballHandler),
                        new WaitCommand(1.5)
                ),
                race(
                        new Shoot(ballHandler),
                        new StayOnTarget(map, drive, vision),
                        new WaitCommand(4)
                )
        );
        addRequirements(ballHandler);
    }
}

File: westcoast/src/main/kotlin/frc/commands/ballhandling/Shooting.kt
package frc.commands.ballhandling

import edu.wpi.first.wpilibj.drive.DifferentialDrive
import edu.wpi.first.wpilibj2.command.SequentialCommandGroup
import edu.wpi.first.wpilibj2.command.WaitCommand
import frc.robot.maps.RobotMap
import frc.subsystems.BallHandlingSubsystem
import frc.subsystems.VisionRunner
import frc.subsystems.targeting.AlignToTarget
import frc.subsystems.targeting.StayOnTarget

class Shooting(map: RobotMap?, drive: DifferentialDrive?, vision: VisionRunner?, ballHandler: BallHandlingSubsystem) : SequentialCommandGroup() {
  init {
    addCommands(
        AlignToTarget(map!!, drive!!, vision!!),
        race(
            ShooterAccelerator(ballHandler),
            WaitCommand(1.5)
        ),
        race(
            Shoot(ballHandler),
            StayOnTarget(map, drive, vision),
            WaitCommand(4.0)
        )
    )
    addRequirements(ballHandler)
  }
}